src/strategies/staked-uniswap/index.ts

```diff
@@ -0,0 +1,76 @@
+import { multicall } from '../../utils';
+import { abi as uniAbi } from './UniV2.json';
+
+export const author = 'vfatouros';
+export const version = '0.1.0';
+
+const tokenAbi = [
+  {
+    constant: true,
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'account',
+        type: 'address'
+      }
+    ],
+    name: 'balanceOf',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256'
+      }
+    ],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
+
+export async function strategy(
+  _space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const res = await Promise.all([
+    multicall(
+      network,
+      provider,
+      uniAbi,
+      [[options.uniswapAddress, 'totalSupply', []]],
+      { blockTag }
+    ),
+    multicall(
+      network,
+      provider,
+      tokenAbi,
+      [[options.tokenAddress, 'balanceOf', [options.uniswapAddress]]],
+      { blockTag }
+    ),
+    multicall(
```

Comment:
It would be great if you can make a single multicall to gather all info you need at once. This way it will only have a single request to be sent instead of 3, it's more light for the users browsers.

Reply:
Yes I should have looked at multicall a bit closer! That is resolved now.